expan/core/experiment.py

```diff
@@ -339,3 +387,20 @@ def run_goodness_of_fit_test(self, observed_freqs, expected_freqs, alpha=0.01, m
             raise ValueError("Variant split check was cancelled since observed or expected frequencies "
                              "are less than 2.")
         return split_is_unbiased, p_value
+
+def _choose_threshold_type(data):
+    """ Heuristics used to decide what filtering method to use."""
+    assert len(data), 'data should be non-empty'
+    assert data.min,  'data should have min method'
+    assert data.max,  'data should have max method'
+    min, max = data.min(), data.max()
+
+    if min < 0.0 and max > 0.0:
+        return 'two-sided'
+    elif max > 0.0:
+        return 'upper'
+    elif min < 0.0:
+        return 'lower'
+    else:
+        # it doesn't really matter since in this case min == max == 0.0
```

Comment:
If there are any `NaN`s in the data, it will reach this `else`. So I think the comment is incorrect

It's a bit awkward here though. I would suggest `data.min(skipna=True)` as this works for `pandas.Series`, but it doesn't work for `numpy` arrays. So I don't know what types and methods we can rely on here

Ideally, we would do one of these things in this `_choose_threshold_type` function:
 - remove the nan
 - or replace the nans with zero

But, as this is just the default behaviour for columns without an explicitly-requested threshold_type, maybe we don't need to worry about this

Reply:
done, sticking to `Series`.

Reply:
Nice. Resolving.